### PR TITLE
Add a `.optional.` file name flag for optional behavior

### PR DESCRIPTION
**Pull request by foolip**
Fixes https://github.com/web-platform-tests/wpt/issues/12766.

### CODE REVIEW

**Review by sideshowbarker — APPROVED** [review]

File: docs/_writing-tests/file-names.md [hunk 1]
```diff
@@ -54,6 +54,11 @@ themselves precede any test type flag, but are otherwise unordered.
  : (js files only) Indicates that the file generates tests in which it
     is [run in multiple scopes][multi-global-tests].
 
+`.optional`
+ : Indicates that a test makes assertions about optional behavior in a
+   specification, typically marked by the [RFC 2119] "MAY" or "OPTIONAL"
```
Comment (domenic) [inline docs/_writing-tests/file-names.md]: What about "should"?

Comment (foolip) [reply docs/_writing-tests/file-names.md]: I think in practice we're just adding tests for such requirements, or at least I've never seen a discussion about whether it's okay to add such tests.

Comment (domenic) [reply docs/_writing-tests/file-names.md]: Hmm, for some reason I thought that's what spawned this discussion. Maybe call that out explicitly here, to reduce confusion?

Comment (foolip) [reply docs/_writing-tests/file-names.md]: I've added a bit about should and must.

**Review by zcorpan — APPROVED** [review]

**Review by Ms2ger — APPROVED** [review]

File: docs/_writing-tests/file-names.md (hunk 1 above)
Comment (gsnedders) [reply docs/_writing-tests/file-names.md]: That was my understanding of @jgraham's view (that adding such tests often has little benefit).

File: docs/_writing-tests/file-names.md [hunk 2]
```diff
@@ -54,6 +54,12 @@ themselves precede any test type flag, but are otherwise unordered.
...
+   keywords. This flag should not be used for "SHOULD", such requirements
```
Comment (domenic) [inline docs/_writing-tests/file-names.md]: Nit: comma -> semicolon

**Review by domenic — APPROVED** [review]

**Pull request merged**